includes/Builders/Prompt_Builder.php

```diff
@@ -0,0 +1,176 @@
+ * @method bool is_supported_for_speech_generation() Checks if the prompt is supported for speech generation.
+ * @method bool is_supported_for_music_generation() Checks if the prompt is supported for music generation.
+ * @method bool is_supported_for_embedding_generation() Checks if the prompt is supported for embedding generation.
+ * @method GenerativeAiResult generate_result(?CapabilityEnum $capability = null) Generates a result from the prompt.
+ * @method GenerativeAiResult generate_text_result() Generates a text result from the prompt.
+ * @method GenerativeAiResult generate_image_result() Generates an image result from the prompt.
+ * @method GenerativeAiResult generate_speech_result() Generates a speech result from the prompt.
+ * @method GenerativeAiResult convert_text_to_speech_result() Converts text to speech and returns the result.
+ * @method string generate_text() Generates text from the prompt.
+ * @method list<string> generate_texts(?int $candidateCount = null) Generates multiple text candidates from the prompt.
+ * @method File generate_image() Generates an image from the prompt.
+ * @method list<File> generate_images(?int $candidateCount = null) Generates multiple images from the prompt.
+ * @method File convert_text_to_speech() Converts text to speech.
+ * @method list<File> convert_text_to_speeches(?int $candidateCount = null) Converts text to multiple speech outputs.
+ * @method File generate_speech() Generates speech from the prompt.
+ * @method list<File> generate_speeches(?int $candidateCount = null) Generates multiple speech outputs from the prompt.
+ *
+ * @phpstan-import-type MessageArrayShape from Message
+ * @phpstan-import-type MessagePartArrayShape from MessagePart
+ *
+ * @phpstan-type Prompt string|MessagePart|Message|MessageArrayShape|list<string|MessagePart|MessagePartArrayShape>|list<Message>|null
+ */
+class Prompt_Builder {
+
+	/**
+	 * Wrapped prompt builder instance.
+	 *
+	 * @since n.e.x.t
+	 * @var PromptBuilder
+	 */
+	private PromptBuilder $builder;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param ProviderRegistry $registry The provider registry for finding suitable models.
+	 * @param Prompt           $prompt   Optional initial prompt content.
+	 */
+	public function __construct( ProviderRegistry $registry, $prompt = null ) {
+		$this->builder = new PromptBuilder( $registry, $prompt );
+	}
+
+	/**
+	 * Magic method to proxy snake_case method calls to their PHP AI Client camelCase counterparts.
+	 *
+	 * This allows WordPress developers to use snake_case naming conventions.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string            $name      The method name in snake_case.
+	 * @param array<int, mixed> $arguments The method arguments.
+	 * @return mixed The result of the parent method call.
+	 */
+	public function __call( string $name, array $arguments ) {
+		$callable = $this->get_builder_callable( $name );
+		return call_user_func_array( $callable, $arguments );
```

Comment:
Micro optimization nit:

```suggestion
		return $callable( ...$arguments );
```

Reply:
Does that syntax work on any callable, including `array( $instance, 'some_method' )`?

Reply:
Yup!